### PR TITLE
[liblas] Fix geotiff dependency, document usage, add feature 'tools'

### PR DESCRIPTION
**Pull request by dg0yt**
Alternative to #25636.
Includes #25354.

- #### What does your PR fix?
  Adds GeoTIFF package lookup to the exported config.
  Adds a valid `usage` file.
  Adds a `tools` feature for building utility programs.
  Cleanup and modernization.
  **liblas always builds shared libraries.** This means `x64-windows` is actually unsupported. **This PR doesn't change that.** It just ensure that the DLLs are retained for `x64-windows-static-md`.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  unchanged, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

### CODE REVIEW

File: ports/liblas/vcpkg.json [hunk 1]
```diff
@@ -1,8 +1,10 @@
 {
   "name": "liblas",
   "version": "1.8.1",
-  "port-version": 9,
+  "port-version": 10,
   "description": "A C/C++ library for reading and writing the very common LAS LiDAR format.",
+  "license": null,
```
Comment (dg0yt) [inline ports/liblas/vcpkg.json:6]: Cf. https://tracker.debian.org/media/packages/libl/liblas/copyright-1.8.1-10

File: ports/liblas/fix-cmake-config.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/cmake/liblas-config.cmake.in b/cmake/liblas-config.cmake.in
+index 175e997..8a55804 100644
+--- a/cmake/liblas-config.cmake.in
++++ b/cmake/liblas-config.cmake.in
+@@ -19,6 +19,8 @@ set (libLAS_INCLUDE_DIRS "${PROJECT_ROOT_DIR}/include")
+ set (libLAS_LIBRARY_DIRS "${PROJECT_ROOT_DIR}/lib")
+ set (libLAS_BINARY_DIRS "${PROJECT_ROOT_DIR}/bin")
+ 
++include(CMakeFindDependencyMacro)
++find_dependency(GeoTIFF CONFIG)
```
Comment (JackBoosY) [inline ports/liblas/fix-cmake-config.patch:10]:
```cmake
find_package (GeoTIFF 1.3.0 )
if(GEOTIFF_FOUND)
    set(WITH_GEOTIFF TRUE)
```

So should we use condition `if (@WITH_GEOTIFF@)` here?

File: ports/liblas/portfile.cmake [hunk 3]
```diff
@@ -1,3 +1,5 @@
+vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
```
Comment (JackBoosY) [inline ports/liblas/portfile.cmake:1]: Yes, it's only shared:
```cmake
add_library(${LIBLAS_LIB_NAME} SHARED ${LIBLAS_SOURCES})
add_library(${LIBLAS_C_LIB_NAME} SHARED ${LIBLAS_C_SOURCES})
```

File: ports/liblas/usage [hunk 4]
```diff
@@ -0,0 +1,9 @@
+liblas provides CMake targets:
+
+    # C API
+    find_package(liblas CONFIG REQUIRED)
```
Comment (JackBoosY) [inline ports/liblas/usage:4]: It's incorrect.
In _liblas-config-version.cmake.in_:
```cmake
if (NOT PACKAGE_FIND_NAME STREQUAL "@PROJECT_NAME@")
```
In _CMakeLists.txt_:
```cmake
project(libLAS)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/liblas/fix-cmake-config.patch (hunk 2 above)
Comment (dg0yt) [reply ports/liblas/fix-cmake-config.patch:10]: No. GeoTIFF is required.

File: ports/liblas/usage (hunk 4 above)
Comment (dg0yt) [reply ports/liblas/usage:4]: Bad practice, breaking `find_package` flexibility. But this package is EOL, so I will adjust the usage document.

File: ports/liblas/usage [hunk 5]
```diff
@@ -0,0 +1,9 @@
+liblas provides CMake targets:
+
+    # C API
+    find_package(libLAS CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE las_c)
```
Comment (JackBoosY) [inline ports/liblas/usage:5]:
```cmake
# Create imported target liblas
add_library(liblas SHARED IMPORTED)

# Create imported target liblas_c
add_library(liblas_c SHARED IMPORTED)
```
Can you double confirm the target names?

Comment (dg0yt) [reply ports/liblas/usage:5]: I see, IIRC it depends on `WIN32`.

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
Then I will revert liblas part in my PR https://github.com/microsoft/vcpkg/pull/25636 later.

**Pull request merged**